packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -110,6 +111,8 @@ class TestCommand extends FastFlutterCommand {
     if (shouldRunPub) {
       await pubGet(context: PubContext.getVerifyContext(name), skipPubspecYamlCheck: true);
     }
+    await _buildTestAsset();
```

Comment:
Does this add measurable overhead when running the test over and over again? If so, we could potentially reuse the existing pubGet check, make it return a bool for whether or not the pubspec.yaml file had updated. Only if the pubspec file is updated, or if the asset directory doesn't exist, do we need to re-run ot

Reply:
This will only run once within a flutter test. The overhead is depends on the assets size.
I check the code in flutter run. the run cold will build it everytime, while run hot will check if the pubspec.yaml change or not. I think it is ok to always rebuild during test. if we want to be smart, we not only need to check the pubspec.yaml, we also need to check ant of assets files changed or not.

Reply:
Actually, thinking about it more - we can't really do this, since a user may edit an asset which would then not be copied.

We should expose a flag in the test command to disable this functionality though

Reply:
you mean expose a flag to skip rebuilding asset?

Reply:
yes, similar to how we have a `--pub,--no-pub` flag.

Reply:
done